libs/timeseries/SyntheticTimeSeriesCreator.h

```diff
@@ -100,7 +100,7 @@ namespace mkc_timeseries
 	// correctly for this bar -- tested this on KC and MSFT data to make sure
 	if(timeFrameId == 1)
 	  syntheticTimeSeries->addEntry(OHLCTimeSeriesEntry<Decimal>(
-								     mEntryDate, mOpen, mHigh, mLow, mClose, mOriginalHourlyTimeSeries->getVolumeValue(mOriginalHourlyTimeSeries->beginRandomAccess(), 0), TimeFrame::DAILY));
+								     mEntryDate, mOpen, mHigh, mLow, mClose, mOriginalHourlyTimeSeries->beginRandomAccess()->getVolumeValue(), TimeFrame::DAILY));
```

Comment:
I am guessing `beginRandomAccess()` is never `end` at that stage?

Reply:
It could be. This could is unlikely to be used again. Even it if is it needs a thorough review and reworking.